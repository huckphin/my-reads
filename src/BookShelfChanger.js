import React, { Component } from 'react';
import PropTypes from 'prop-types';

class BookShelfChanger extends Component {
    static propTypes = {
        book: PropTypes.object.isRequired,
        onUpdateBook: PropTypes.func.isRequired
    }

    render () {
        const { book, onUpdateBook } = this.props;

        return (
            <div className="book-shelf-changer">
                <select onChange={(event) => onUpdateBook(book, event.target.value)} defaultValue="move-to">
                    <option value="move-to">Move to...</option>
                    <option value="currentlyReading">Currently Reading</option>
                    <option value="wantToRead">Want to Read</option>
                    <option value="read">Read</option>
                    <option value="none">None</option>
                </select>
            </div>
        )
    }
};

export default BookShelfChanger;