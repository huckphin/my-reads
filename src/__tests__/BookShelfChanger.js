import React from 'react';
import BookShelfChanger from '../BookShelfChanger';
import renderer from 'react-test-renderer';

test('will render a BookShelfChanger', () => {
    const component = renderer.create(
        <BookShelfChanger />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});