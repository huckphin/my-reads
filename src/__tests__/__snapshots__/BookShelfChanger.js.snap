// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render a BookShelfChanger 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    onChange={[Function]}
  >
    <option
      selected="selected"
      value="move-to"
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;
